flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/utils/TiKVTypeUtils.java

```diff
@@ -233,17 +280,40 @@ public static DataType getFlinkType(final org.tikv.common.types.DataType dataTyp
             if (objects[i] == null) {
                 continue;
             }
+            org.tikv.common.types.DataType tidbType = tableInfo.getColumn(i).getType();
+            DataType flinkType = getFlinkType(tidbType);
             objects[i] =
-                    toRowDataType(
-                            getObjectWithDataType(
-                                            objects[i],
-                                            getFlinkType(tableInfo.getColumn(i).getType()))
-                                    .get(),
-                            getFlinkType(tableInfo.getColumn(i).getType()));
+                    toRowDataType(getObjectWithDataType(objects[i], flinkType).get(), flinkType);
+            if (tidbType.isUnsigned()) {
+                objects[i] = dealUnsignedColumnValue(tidbType, objects[i]);
+            }
         }
         return objects;
     }
 
+    /** Deal with unsigned column's value. */
+    public static Object dealUnsignedColumnValue(
+            org.tikv.common.types.DataType dataType, Object object) {
+        switch (dataType.getType()) {
+            case TypeTiny:
+                return Short.valueOf((short) Byte.toUnsignedInt(((Short) object).byteValue()));
+            case TypeShort:
+                return Integer.valueOf(Short.toUnsignedInt(((Integer) object).shortValue()));
+            case TypeInt24:
+                return Integer.valueOf(((int) object) & 0xffffff);
+            case TypeLong:
+                return Long.valueOf(Integer.toUnsignedLong(((Long) object).intValue()));
+            case TypeLonglong:
+                return DecimalData.fromBigDecimal(
```

Comment:
Could you post the related tikv link here for let other developers know more?

Reply:
Ok, I will add link https://docs.pingcap.com/zh/tidb/stable/data-type-numeric#%E5%AD%98%E5%82%A8%E7%A9%BA%E9%97%B4%E4%BB%A5%E5%8F%8A%E5%8F%96%E5%80%BC%E8%8C%83%E5%9B%B4 here.

Reply:
English doc is more recommended, https://docs.pingcap.com/tidb/stable/data-type-numeric